src/poetry/installation/pip_installer.py

```diff
@@ -56,6 +56,8 @@ def install(self, package: Package, update: bool = False) -> None:
                     f" {parsed.hostname}</warning>"
                 )
                 args += ["--trusted-host", parsed.hostname]
+            elif repository.trusted:
+                args += ["--trusted-host", parsed.hostname]
```

Comment:
This only affects the old installer as the new installer downloads the packages via the authenticator.